example/generator/odp_generator.c

```diff
@@ -838,39 +892,55 @@ static int gen_recv_thread(void *arg)
 		if (thr_args->stop)
 			break;
 
-		/* Use schedule to get buf from any input queue */
-		ev_cnt = odp_schedule_multi(NULL, ODP_SCHED_NO_WAIT,
-					    events, burst_size);
-		if (ev_cnt == 0)
-			continue;
-		for (i = 0, pkt_cnt = 0; i < ev_cnt; i++) {
-			pkt = odp_packet_from_event(events[i]);
-			itf = &itfs[odp_pktio_index(odp_packet_input(pkt))];
-
-			if (odp_packet_has_ipv4(pkt)) {
-				if (itf->config.pktin.bit.ipv4_chksum) {
-					if (odp_packet_has_l3_error(pkt))
-						printf("HW detected L3 error\n");
-				}
-			}
+		pkt_cnt = odp_pktin_recv_tmo(pktin, pkts, burst_size,
+					     ODP_PKTIN_NO_WAIT);
```

Comment:
why not ODP_PKTIN_WAIT?

Reply:
and why odp_pktin_recv_tmo() and not odp_pktin_recv() ?

Reply:
Agree with @muvarov, this could use some comments to explain why these calls are being used. You'd expect a dedicated RX thread to simply wait for packet input.

Reply:
yep. My impression was that odp_pktin_recv() is blocking but it seems is no: in 'ping' mode we need to be able to stop the receive thread after a number of pings.

Reply:
it's not blocking, it just polling function.